bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -104,27 +106,30 @@ private void initModel() {
         navPageSize = properties.get(PN_PAGE_SIZE, currentStyle.get(PN_PAGE_SIZE, PAGE_SIZE_DEFAULT));
         loadClientPrice = properties.get(PN_LOAD_CLIENT_PRICE, currentStyle.get(PN_LOAD_CLIENT_PRICE, LOAD_CLIENT_PRICE_DEFAULT));
 
-        setNavPageCursor();
+        String currentPageIndexCandidate = request.getParameter(CURRENT_PAGE_QUERY_STRING);
+        // make sure the current page from the query string is reasonable i.e. numeric and over 0
+        Integer currentPageIndex = calculateCurrentPageCursor(currentPageIndexCandidate);
+
+        Map<String, String> searchFilters = createFilterMap(request.getParameterMap());
 
         // get product template page
         productPage = SiteNavigation.getProductPage(currentPage);
         if (productPage == null) {
             productPage = currentPage;
         }
 
-        locale = productPage.getLanguage(false);
-
         MagentoGraphqlClient magentoGraphqlClient = MagentoGraphqlClient.create(resource);
 
         // Parse category id from URL
-        final String categoryId = parseCategoryId();
+        categoryId = parseCategoryId(this.request.getRequestPathInfo().getSelectorString(), request.getParameter(
+            "category_id"));
```

Comment:
Why do you also check the `category_id` parameter from the URL? I saw that the filters on the category pages include that `category_id` request parameter, why do we need this? (since we anyway get it from the url selector)

I'm asking because we will have a merge conflict here with https://github.com/adobe/aem-core-cif-components/pull/248 so I'd like to fully understand if this is needed or not.

Reply:
It was quite a while ago I wrote this and I'm not sure if at the time I added this as a fallback so the query string parameters alone could control all of the options, but that said I can remove this.

That said, as it is currently in `master` there is a null check for this value but not for integer value / etc so this was partially why I refactored to return an `Optional`. For now I will revert to the version you have in `master` though with the understanding this is improved / updated in the PR you referenced!